key.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2022 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package stoabs
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+)
+
+// Key is an abstraction for a key in a key/value pair. The underlying implementation determines if the string or byte representation is used.
+type Key interface {
+	fmt.Stringer
+
+	// Bytes returns a byte representation of this key
+	Bytes() []byte
+	// Next returns the next logical key. The next for the number 1 would be 2. The next for the string "a" would be "b"
+	Next() Key
+}
+
+// Uint32Key is a type helper for a uint32 as Key
+type Uint32Key uint32
+
+func (u Uint32Key) String() string {
+	return fmt.Sprintf("%d", u)
+}
+
+// Bytes outputs the byte representation of the uint32 in BigEndian
+func (u Uint32Key) Bytes() []byte {
+	bytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(bytes[:], uint32(u))
+	return bytes
+}
+
+func (u Uint32Key) Next() Key {
+	return u + 1
+}
+
+// Sha256Key is a type helper for a 256 bits hash as Key
```

Comment:
You could actually leave out "SHA" here and call it `256BitKey` (since there's nothing SHA about it at this point)

Reply:
You can't call it that, can't start with numbers. I'll call it a HashKey then.